app/Services/MarketDataProviders/CoinGecko.php

```diff
@@ -77,6 +77,11 @@ public function priceAndPriceChange(string $baseCurrency, Collection $targetCurr
     {
         $data = Http::get('https://api.coingecko.com/api/v3/coins/'.Str::lower($baseCurrency))->json();
 
+        if ($this->isEmptyResponse($data)) {
+            /** @var Collection<string, MarketData> */
+            return collect([]);
```

Comment:
will this result in storing an empty response and showing `0` in the frontend or does this make it use older data?

Reply:
Passing the empty array you've highlighted results in the command not updating any prices since there are no values to loop over. So the 2 lines highlighted in the screenshot below never run because of an empty array

![image](https://user-images.githubusercontent.com/8069294/184239498-54cd8fd2-5c7e-4f4c-b7ca-fe86aa04c2fd.png)

Reply:
I've added a test to make sure prices don't get reset to zero on coingecko being down